de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/bpartner/service/impl/BPartnerStatsDAO.java

```diff
@@ -150,7 +150,10 @@ public BigDecimal retrieveSOCreditUsed(@NonNull final BPartnerStats bpStats)
 				+ " AND p.C_Charge_ID IS NULL AND p.DocStatus IN ('CO','CL')),0)*(-1), "
 				// open invoice candidates
 				+ "COALESCE((SELECT SUM(currencyBase(ic.LineNetAmt,ic.C_Currency_ID,ic.DateOrdered, ic.AD_Client_ID,ic.AD_Org_ID)) FROM C_Invoice_Candidate ic "
-				+ "WHERE ic.Bill_BPartner_ID=bp.C_BPartner_ID AND ic.Processed='N'),0) "
+				+ "WHERE ic.Bill_BPartner_ID=bp.C_BPartner_ID AND ic.Processed='N' AND ic.IsSOTrx='Y'),0), "
+				// tax for open invoice candidates
+				+ "COALESCE((SELECT SUM(currencyBase(invoicecandidatetaxamt(ic.C_Invoice_Candidate_ID),ic.C_Currency_ID,ic.DateOrdered, ic.AD_Client_ID,ic.AD_Org_ID)) FROM C_Invoice_Candidate ic "
+				+ "WHERE ic.Bill_BPartner_ID=bp.C_BPartner_ID AND ic.Processed='N' AND ic.IsSOTrx='Y'),0) "
```

Comment:
this change looks very inefficient. Pls consider following:
* extract it to a view => would make it much more readable and obvious
* instead of joining the C_Invoice_Candidate table 3 times (2 times in the query above and one time in invoiceCandidateTaxAmt function) consider calculating everything in one step.